tests/pass/shims/fs.rs

```diff
@@ -27,7 +28,10 @@ fn main() {
     test_file_sync();
     test_errors();
     test_rename();
+    // solarish needs to intercept readdir/readdir64 for these tests.
+    #[cfg(not(any(target_os = "solaris", target_os = "illumos")))]
     test_directory();
+    #[cfg(not(any(target_os = "solaris", target_os = "illumos")))]
     test_canonicalize();
```

Comment:
Please use `if cfg!(...)` instead of `#[cfg]`, then you can avoid making the import so messy.

Reply:
but what would be the condition tough ?

Reply:
`if cfg!(not(any(target_os = "solaris", target_os = "illumos"))) { ... }`

Reply:
but that s the thing, I tried this and the tests are called still on my illumos instance.

Reply:
ok will give it a go in few hours.

Reply:
so, yes that was an error on my part indeed :) let s see how it goes ..